### PR TITLE
Added method to get components registered to the Core. Updated rotriever.lock file.

### PR DESCRIPTION
**Pull request by Phalanxia**
Adds getRegisteredComponents to the Core. Updates rotriever.lock file to the latest format.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,5 @@
 *.rbxlx
 *.rbxm
 *.rbxmx
-packages
\ No newline at end of file
+packages
+TestEZ
```
Comment (AmaranthineCodices) [inline .gitignore:6]: this change is unnecessary

File: src/Core.lua [hunk 2]
```diff
@@ -220,6 +220,17 @@ function Core:getComponentClass(className)
     return componentClass
 end
 
+--[[
+
+    Returns all components classes registered to the in the Core.
```
Comment (AmaranthineCodices) [inline src/Core.lua:225]:
```suggestion
    Returns all components classes registered in the Core.
```

File: src/Core.lua [hunk 3]
```diff
@@ -220,6 +220,17 @@ function Core:getComponentClass(className)
...
+
+    Do not mutate the returned value of this function.
+
+]]
+function Core:getRegisteredComponents()
```
Comment (AmaranthineCodices) [inline src/Core.lua:230]: A comment about how this table is laid out would be useful (right now it's a map of name -> class).

File: src/Core.spec/init.lua [hunk 4]
```diff
@@ -52,6 +52,21 @@ return function()
...
+            name = "TestComponent",
+            generator = function()
+                return {}
+            end
+        })
+
+        it("should succeed when called", function()
+            local core = Core.new()
+            core:registerComponent(ComponentClass)
+            core:getRegisteredComponents()
```
Comment (AmaranthineCodices) [inline src/Core.spec/init.lua:66]: We should check that it actually returns the component class and not just an empty table!

File: src/Core.spec/init.lua [hunk 5]
```diff
@@ -52,6 +52,21 @@ return function()
         end)
     end)
 
+    describe("getComponents", function()
```
Comment (AmaranthineCodices) [inline src/Core.spec/init.lua:55]: Should be
```suggestion
    describe("getRegisteredComponents", function()
```

File: src/Core.spec/init.lua [hunk 6]
```diff
@@ -52,6 +52,21 @@ return function()
...
+        local ComponentClass = defineComponent({
+            name = "TestComponent",
+            generator = function()
+                return {}
+            end
+        })
+
+        it("should succeed when called", function()
```
Comment (AmaranthineCodices) [inline src/Core.spec/init.lua:63]: With the below suggestion this should be something like "should return registered components".

**Comment by Phalanxia** [comment]
OOPS

**Pull request merged**